cardinal/world.go

```diff
@@ -254,6 +254,9 @@ func (w *World) IsGameRunning() bool {
 }
 
 func (w *World) ShutDown() error {
+	if w.cleanup != nil {
+		w.cleanup()
+	}
 	ok := w.gameSequenceStage.CompareAndSwap(gamestage.StageRunning, gamestage.StageShuttingDown)
```

Comment:
This is a bug. Cleanup can contain instructions to close miniredis. The logic write below this comment actually on occasion can stop the cleanup from ever being called which means that some tests left the port used.